spec/requests/schedule_spec.rb

```diff
@@ -0,0 +1,148 @@
+# frozen_string_literal: true
+
+RSpec.describe "Schedule" do
+  it_behaves_like "requests that assert edition state",
+                  "scheduling a non editable edition",
+                  routes: { new_schedule_path: %i[get post] } do
+    let(:edition) { create(:edition, :scheduled) }
+  end
+
+  it_behaves_like "requests that assert edition state",
+                  "scheduling an edition without a proposed publish time",
+                  routes: { new_schedule_path: %i[get post] } do
+    let(:edition) { create(:edition, proposed_publish_time: nil) }
+  end
+
+  it_behaves_like "requests that assert edition state",
+                  "managing the scheduling of a non-scheduled edition",
+                  routes: { edit_schedule_path: %i[get patch],
+                            schedule_path: %i[delete],
+                            scheduled_path: %i[get] } do
+    let(:edition) { create(:edition, :published) }
+  end
+
+  describe "GET /documents/:document/schedule/new" do
+    it "redirects to document summary with errors when the content isn't publishable" do
```

Comment:
This seemed a bit overkill, but then I noticed that it's [missing from the interactor spec](https://github.com/alphagov/content-publisher/blob/master/spec/interactors/schedule/update_interactor_spec.rb#L71). Could we plug the hole there, and just have one spec to focus on "issues" here? It's pretty minor, really - the testing strategy was tricky to figure out, since the corresponding feature test has three cases, and here we only have two.

Reply:
I have a bit of confusion here as you're linking to the update interactor but this is the new endpoint. I also missed the success case test, which I've since added.

But anyway on the new endpoint there are three responses the controller can return and there are now three tests which cover each.

Reply:
Oops - you're right. I *was* looking at the wrong thing!